### PR TITLE
refactor(app_config): Using ConfigBuilder instead of merging over Config

### PR DESCRIPTION
**Pull request by Nindouja**
tl;dr: saw some warnings while building a fresh copy of the repo, so I decided to try and fix those. Not really confident it improves the big picture, tho.

**For some context**, I started using rust-starter to practice during my learning process. I figured a simple CLI app is a good first project. **But that makes me a total noob**, so please keep that in mind while reviewing.
When I first started the blank project I noticed some warnings but didn't pay much attention to them. Now I start building my CLI app, I'd like to remove those, so I edit some of the app_config file to get rid of some.

**The actual change** consists of using a config builder in the RwLock instead of the raw config. This get rid of all the deprecation warnings on the .merge(). The config is only built once .fetch() or .get() are called.

**Probable improvements** but due to my lack of experience, I'm not sure how to fix them.
* Remove .unwrap() line 78 and 90 (on the locks), 
* Maybe the .clone() are not really elegant either

### CODE REVIEW

**Comment by Nindouja** [comment]
I've added two other commits for others warnings. I don't know if I should open dedicated PR for such little things
